### DISCUSSION TITLE
Conditional XAML for menu items

### DISCUSSION THREAD

**Discussion by LilyStilson** [AvaloniaUI/Avalonia#14519] [Q&A]
I have an application that by design is supposed to have a menu item with app name on both supported platforms (Windows and macOS).
![image](https://github.com/AvaloniaUI/Avalonia/assets/42270214/e305e83c-816d-496b-a307-95927b78e7cc)

I managed to achieve this by adding same menu items both in `MainWindow.axaml` and `App.axaml`. While it's not a problem on Windows, since that first native menu item is being ignored, it is a problem on macOS as both menu items are being displayed.
![image](https://github.com/AvaloniaUI/Avalonia/assets/42270214/3a8437bc-0abf-45bb-b106-d57f1f879f69)
(also I just noticed that menu items after Edit are not being displayed on macOS, despite being present in XAML...)

Is there any way to hide that second menu item on macOS or make first menu item appear on Windows in `MainWindow` without actually making two copies of same menu items?

I also should mention that I've tried methods described in #1434, none worked for `<NativeMenu />` and `<NativeMenuItem />`.

**Comment by thevortexcloud** [comment]
Did you try OnPlatform?

https://docs.avaloniaui.net/docs/guides/platforms/platform-specific-code/xaml

**Reply by LilyStilson** [reply]
> I also should mention that I've tried methods described in https://github.com/AvaloniaUI/Avalonia/issues/1434

Of course I did try it, as it is the main topic of #1434. It does not work for `<NativeMenu />`. Same as `<NativeMenu />` has no `IsVisible` property, only `IsEnabled` and setting this one to `False` on macOS does not seem to do anything.
```xaml
<NativeMenuItem Header="AErender Launcher" IsEnabled="{OnPlatform macOS='False', Default='True'}" />
```
